crates/re_data_source/src/data_loader/loader_external.rs

```diff
@@ -185,6 +190,39 @@ impl crate::DataLoader for ExternalLoader {
                     }
                 };
 
+                // We have to wait in order to know whether the child process is a compatible loader.
+                //
+                // This can manifest itself in two distinct ways:
+                // 1. If it exits immediately with an INCOMPATIBLE exit code, then we have our
+                //   answer straight away.
+                // - If it starts streaming data, then we immediately assume it's compatible.
+                loop {
+                    match child.try_wait() {
+                        Ok(Some(_)) => break,
+                        Ok(None) => {
+                            // The child process has started streaming data, it is therefore compatible.
+                            // Let's get out ASAP.
+                            if rx_is_sending_data.try_recv().is_ok() {
+                                re_log::debug!(loader = ?exe, ?filepath, "compatible external loader found");
+                                tx_feedback.send(CompatibleLoaderFound).ok();
+                                break; // we still want to check for errors once it finally exits!
+                            }
+
+                            match rayon::yield_now() {
+                                Some(rayon::Yield::Executed) => {}
+                                _ => std::thread::yield_now(),
+                            }
```

Comment:
This loop is effectively a busy-loop until the first data is sent from it, correct? Might be worth pointing that out, and maybe have a profiling scope just for the loop (`profile_scooe!("busy-wait-for-plugin")` or whatever)